WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsActivity.java

```diff
@@ -80,366 +26,45 @@ public void onCreate(Bundle savedInstanceState) {
-                                                mNotesListFragment.setNoteIsHidden(note.getId(), false);
-                                                ToastUtils.showToast(NotificationsActivity.this,
-                                                        R.string.error_moderate_comment,
-                                                        ToastUtils.Duration.LONG
-                                                );
-                                            }
-                                        }
-                                    });
-                        }
-
-                        @Override
-                        public void onClear(@Nonnull Parcelable[] token) {
-                            //noop
-                        }
-
-                        @Override
-                        public void onUndo(Parcelable parcelable) {
-                            mNotesListFragment.setNoteIsHidden(note.getId(), false);
-                        }
-                    }).show();
-        }
-    }
-
-    @Override
-    public void onCommentChanged(CommentActions.ChangedFrom changedFrom, CommentActions.ChangeType changeType) {
-        // pop back stack if we edited a comment notification, so simperium will show the change
-        if (changedFrom == CommentActions.ChangedFrom.COMMENT_DETAIL
-                && changeType == CommentActions.ChangeType.EDITED) {
-            FragmentManager fm = getFragmentManager();
-            if (fm.getBackStackEntryCount() > 0) {
-                fm.popBackStack();
-            }
-        }
-    }
-
-    private class NoteClickListener implements NotificationsListFragment.OnNoteClickListener {
-        @Override
-        public void onClickNote(String noteId) {
-            if (TextUtils.isEmpty(noteId)) return;
-
-            // open the latest version of this note just in case it has changed - this can
-            // happen if the note was tapped from the list fragment after it was updated
-            // by another fragment (such as NotificationCommentLikeFragment)
-            if (SimperiumUtils.getNotesBucket() != null) {
-                try {
-                    Note note = SimperiumUtils.getNotesBucket().get(noteId);
-                    openNote(note);
-                } catch (BucketObjectMissingException e) {
-                    AppLog.e(T.NOTIFS, "Note could not be found.");
-                }
-            }
-        }
-    }
-
-    // Retrieves comment reply text so we can restore it upon returning to CommentDetailFragment
-    private String getCommentReplyText() {
-        if (mDetailFragment != null && mDetailFragment instanceof CommentDetailFragment) {
-            CommentDetailFragment commentDetailFragment = (CommentDetailFragment)mDetailFragment;
-            if (!TextUtils.isEmpty(commentDetailFragment.getReplyText())) {
-                return commentDetailFragment.getReplyText();
-            }
+        Intent intent = getIntent();
```

Comment:
I love how bare-bones this activity is, but it would be even simpler if these extras could be passed when creating the fragment so no extra logic is necessary (that's going to be important in the ViewPager activity). I usually create a [newInstance](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/2248-notifications-fragment/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderBlogFragment.java#L34) function in the fragment to do this, ex:

```
public static NotificationsListFragment newInstance(String noteId, boolean isReply) {
    Bundle args = new Bundle();
    args.putString(NOTE_ID_EXTRA, noteId);
    args.putBoolean(NOTE_INSTANT_REPLY_EXTRA, isReply);
    NotificationsListFragment fragment = new NotificationsListFragment();
    fragment.setArguments(args);
    return fragment;
}
```